unity/ComputerVision-UnityClient/Assets/Scripts/NetworkManager.cs

```diff
@@ -251,6 +233,7 @@ public void ServerProcessingTimeCalculator(float requestLatency)
                 ServerProcessingTimeRollingAvgQueue.Enqueue(requestLatency);
             }
             avgServerProcessingTime = ServerProcessingTimeRollingAvgQueue.Average();
+            avgNetworkOnlyLatency = (avgFullProcessLatency * 1000) - avgServerProcessingTime;
```

Comment:
This is not an accurate calculation for network-only latency. Network-only latency should basically be a "ping" time, or as close as we can get to that. Where ICMP ping is not available, I have used the time it takes to open a socket as measurement. For WebSocket, I time how long it takes to echo short text payload.

Check out this blog post: https://mobiledgex.com/blog/2020/04/22/computer-vision-websocket-support. Look at the image in the “Results” section.

![image](https://user-images.githubusercontent.com/41596890/93152612-bf327780-f6c4-11ea-9177-935e3f83fad1.png)

In addition to server_processing_time, Full Process also includes the time it takes to upload the image, and the time it takes to receive the results back. That’s why the image size plays such a big roll in the overall latency.

That's why network-only latency can't just be full process latency minus server processing time. The image upload time and results transfer time are discounted in this calculation.

Reply:
Check with @lgarner. He may already have some network latency code in Unity.

Reply:
Actually, I have one more way to measure, and it's what I'm doing for the Javascript REST implementation. Instead of a POST or GET, I do a HEAD request to http://hostname:8008/test/ and measure how long that takes. Not perfect, but as close as I could get to "ping" time with Javascript. See function networkLatency() in https://github.com/mobiledgex/edge-cloud-sampleapps/blob/master/ComputerVisionServer/moedx/static/js/index.js.

Reply:
Thanks a lot @brucearmstrong , Please lmk if you find the new approach better.